schema.ts

```diff
@@ -392,6 +397,84 @@ export interface ScaleSchema extends BaseNumberSchema<ValueHintType.Scale> {
   icon?: ScaleIconSet;
 }
 
+/**
+ * Display types that can be used with an {@link EmailSchema}.
+ */
+export enum EmailDisplayType {
+  /**
+   * Display both icon and email (default).
+   */
+  IconAndEmail = 'iconAndEmail',
+
+  /**
+   * Display icon only.
+   */
+  IconOnly = 'iconOnly',
+
+  /**
+   * Display email address only.
+   */
+  EmailOnly = 'emailOnly',
+}
+
+/**
+ * A schema representing a return value or object property that is an email address.
+ *
+ * An email address can be represented visually as an icon, an icon plus the email address, or
+ * the just the email address.  Autocomplete against previously typed domain names is
+ * also an option in the user interface.
+ */
+export interface EmailSchema extends BaseStringSchema<ValueHintType.Email> {
+  /** Instructs Coda to render this value as an email address. */
+  codaType: ValueHintType.Email;
+  /** How the email should be displayed in the UI. */
+  display?: EmailDisplayType;
+  /** Whether to auto-complete the email domain during user input. */
+  autocomplete?: boolean;
+}
+
+/**
+ * Display types that can be used with a {@link LinkSchema}.
+ */
+export enum LinkDisplayType {
+  /**
+   * Display icon only.
+   */
+  IconOnly = 'IconOnly',
+
+  /**
+   * Display URL.
+   */
+  Url = 'URL',
+
+  /**
+   * Display web page title.
+   */
+  Title = 'Title',
+
+  /**
+   * Display the referenced web page as a card.
+   */
+  Card = 'Card',
+
+  /**
+   * Display the referenced web page as an embed.
+   */
+  Embed = 'Embed',
```

Comment:
hmm not for this PR to address but wondering if this should be consolidated with the Embed type..